### PR TITLE
Remove hard-coded numbers from recovery actions

### PR DESCRIPTION
**Pull request by SteveMacenski**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #775 https://github.com/ros-planning/navigation2/issues/727 |
| Primary OS tested on | CI tests |
| Robotic platform tested on | CI  |

* tested on CI since BT would simply crash if any of the behavior tree stuff was broken. Will test Monday for recovery server side but don't expect issues since primarily is just declaring params or taking new params from action message

---

## Description of contribution in a few bullet points

* got rid of all magic numbers in stack (except timeouts)

---

## Future work that may be required in bullet points

* parameterize timeout

### CODE REVIEW

**Comment by rotu** [comment]
Pity you made this PR too soon. It was almost number 1234

File: nav2_behavior_tree/include/nav2_behavior_tree/spin_action.hpp [hunk 1]
```diff
@@ -31,14 +31,24 @@ namespace nav2_behavior_tree
...
   {
-    goal_.target_yaw = M_PI / 2;
+    double dist;
+    getParam<double>("spin_dist", dist);
+    goal_.target_yaw = dist;
+  }
+
+  static const BT::NodeParameters & requiredNodeParameters()
+  {
+    static BT::NodeParameters params = {{"spin_dist", "1.59"}};
```
Comment (mkhansen-intel) [inline nav2_behavior_tree/include/nav2_behavior_tree/spin_action.hpp]:
```suggestion
    static BT::NodeParameters params = {{"spin_dist", "1.57"}};
```

File: nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml [hunk 2]
```diff
@@ -23,7 +23,7 @@
       <SequenceStar name="RecoveryActions">
         <ClearEntireCostmap service_name="local_costmap/clear_entirely_local_costmap"/>
         <ClearEntireCostmap service_name="global_costmap/clear_entirely_global_costmap"/>
-        <Spin/>
+        <Spin spin_dist="1.59"/>
```
Comment (mkhansen-intel) [inline nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml]:
```suggestion
        <Spin spin_dist="1.57"/>
```

**Comment by SteveMacenski** [comment]
>  Can you change the title to a more explanatory message like "Remove hard-coded numbers from recovery actions"

:crying_cat_face: OK.

**Comment by SteveMacenski** [comment]
On my queue to more thoroughly test today, don't merge for me please

**Comment by rotu** [comment]
@SteveMacenski I know I'm late to the party but I think that spin dist might be better represented as a number of turns than a number of radians. https://en.wikipedia.org/wiki/Turn_(angle)

**Comment by SteveMacenski** [comment]
@rotu I didn't make that up here- its been like this for a bit. I don't think it should be in the scope of this PR. 

And REP103 pretty unambiguously lays out the units: https://www.ros.org/reps/rep-0103.html we should use radians IMO. Do you have an argument for rotations?

**Comment by SteveMacenski** [comment]
Running CI one more time... each time another thing fails and between the 2 they're never consistent so I dont think its my bad

**Comment by SteveMacenski** [comment]
Tested across a bunch of situations and we're all good

**Comment by rotu** [comment]
The reason is because it's not an angular measurement. It's a parameter of the maneuver to be made and it's very likely to be a clean fraction. To me `<Spin spin_dist="1.57">` doesn't express intention as well as `<Spin n_turns="0.25">`. I suppose if we're using REP 103, we may as well use quaternions :-p.

**Comment by SteveMacenski** [comment]
I'm not sure what you mean by that, a quaternion isn't a unit, its a angular representation

**Pull request merged**